File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.js
function solution(arr, k) {
    arr = [...new Set(arr)];
    
    if(arr.length < k) {
        let l = arr.length;
        arr.length = k;
        return arr.fill(-1, l, k);
    } else {
        return arr.slice(0, k);
    }
}

File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int k) {
        int[] result = new int[k];
        Arrays.fill(result, -1);
        
        Set<Integer> set = new HashSet<>();
        
        int i = 0;
        for (int num : arr) {
            if (!set.contains(num)) {
                set.add(num);
                result[i] = num;
                i++;
                
                if (i == k) break;
            }
        }
        
        return result;
    }
}